presto-hive/src/main/java/com/facebook/presto/hive/SyncPartitionMetadataProcedure.java

```diff
@@ -127,8 +132,17 @@ private void doSyncPartitionMetadata(ConnectorSession session, String schemaName
 
         try {
             FileSystem fileSystem = hdfsEnvironment.getFileSystem(context, tableLocation);
-            List<String> partitionsInMetastore = metastore.getPartitionNames(new MetastoreContext(session.getIdentity(), session.getQueryId(), session.getClientInfo(), session.getSource(), getMetastoreHeaders(session), isUserDefinedTypeEncodingEnabled(session), metastore.getColumnConverterProvider()), schemaName, tableName)
+            List<String> partitionNamesInMetastore = metastore.getPartitionNames(metastoreContext, schemaName, tableName)
                     .orElseThrow(() -> new TableNotFoundException(schemaTableName));
+            List<String> partitionsInMetastore = new ArrayList<>(partitionNamesInMetastore.size());
+
+            Map<String, Optional<Partition>> partitionsOptionalMap = metastore.getPartitionsByNames(metastoreContext, schemaName, tableName, partitionNamesInMetastore);
```

Comment:
I had a quick question. Since we just need the list of partition names, why are adding this additional method call to fetch partitions and then creating a new list of partition names? Wouldn't the lists `partitionNamesInMetastore` and `partitionsInMetastore` have the same values?

Is there a mismatch in the output of `metastore.getPartitionNames` and `metastore.getPartitionsByNames`?

Reply:
I actually put down the reasoning in the commit message:

``Current implementation for getting partition names is to take partition values directly
without take into account of the actual path name.
According to the hive source code, this main logic in hive may change that will impact the
 validity of the partition names exists in the storage.
https://github.com/apache/hive/blob/6ada437afc641cd44edf4ea0ccdd905a37385005/standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/utils/FileUtils.java#L246
``

Reply:
Okay, I understand. But we are still not making use of the actual path, in this case, we still just added the values directly using entry.getKey() which will be the same value as what we get from metastore.getPartitionNames. If we want to achieve what the above link suggests, shouldn't we make use of the partition path to generate this value?

Reply:
fixed, good catch, thx

Reply:
Also, I believe we should convert this method call into multiple batches. GlueHiveMetastore implementation already does it but ThriftHiveMetastore will fetch all partitions at once which will result in timeouts or very slow fetches in tables with a considerable number of partitions.

Reply:
+1
Even I noticed now batches are being made only for GlueHiveMetastore implementation but not for other metastores. We can have this call the way we had it earlier or we can extend batch-level implementation even for other megastores like it's being done for Glue?

Reply:
agree, the caller api needs to guard the load not overwhelm the underlying metastore.